logger/logger.go

```diff
@@ -64,6 +64,10 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) Decorate(keyvals ...interface{}) {
+	l.Logger = kitlog.NewContext(l.Logger).With(keyvals...)
```

Comment:
> How we use the logger currently is not thread safe.

Where is the thread safety problem currently (i.e. before my PR)? We use [NewSyncWriter](https://github.com/giantswarm/microkit/blob/master/logger/logger.go#L50) which is [thread safe](https://github.com/go-kit/kit/blob/f66b0e13579bfc5a48b9e2a94b1209c107ea1f41/log/sync.go#L47).

> `s.Log.("cluster-id", clusterID)`

I suppose you mean `s.Log("cluster-id", clusterID)` there.

> So you end up creating a new logger in each method where the cluster ID is being passed.

Yes, this would be an alternative. Basically places that need custom logging decoration would not use the operator's global logger, but instead initialize their own.

To support that, we could add support for `With` stuff to the [`Config`](Jhttps://github.com/giantswarm/microkit/blob/master/logger/logger.go#L50) instead of the `Decorate` method of this PR.

It just makes me sad that `microkit` makes the usage of the underlying package more rigid, and that we have to go hunt for each log line and add the cluster ID instead of doing it in one spot.